indico/modules/auth/controllers.py

```diff
@@ -453,6 +454,7 @@ class Meta:
                 affiliation = fields.String(load_default='')
             phone = fields.String(load_default='')
             affiliation_link = ModelField(Affiliation, data_key='affiliation_id', load_default=None)
+            accept_registration_terms = fields.Bool(required=True)
```

Comment:
this will fail validation when there are no terms set and thus nothing is sent for the field

```suggestion
            if legal_settings.get('register_terms_prompt'):
                accept_registration_terms = fields.Bool(required=True)
```

(untested)

Reply:
Good catch!